docs/toolkits/openstack-operations-maintenance.md

```diff
@@ -1,47 +1,181 @@
+	- Checks for a multiple node cluster:
 SSH to the MYSQL node and log into the database with:
 `mysql`
 and run
 ```
 SHOW GLOBAL STATUS LIKE 'wsrep_%';
 ```
-
 And verify the cluster count matches the number of nodes in your database cluster.
 
-Nova
+	- What MYSQL errors look like
+		If MYSQL is causing issues, you'll usually get a 503 error when attempting to run commands from the CLI.
+	- How to fix
+		This could be as simple as a dead mysql daemon on one of your nodes.  You can diagnose this by running through the checks above.  
+	- Database Discrepancies
+		Throughout the life of your cloud, you may come into issues where the database and openstack reflect different information.  This could be a VM that shows a host on one machine but it's actually on another, a volume that is stuck in a `reserved` state but won't change back to active/available, etc.  To fix this, we need to log into the MYSQL cluster by running `mysql` and updating the discrepancy with `update ITEM set VALUE=DESIRED_VALUE  where id='UUID';`.  You can verify which specific item(s) you'll adjust by adjusting the formula to, `show ITEM where id='UUID';`.
 
+#### Nova
+
+	- Checks:
 To check the nova status, source the Admin openrc file or app credential and run:
 ```
 openstack compute service list
 ```
 
 This will list all nova services and show if any are down or not running.  A "XXX" status simply means that service has missed a heartbeat message in rabbitmq to the control node.  A "down" status means the service has missed enough to be considered off or problematic.
+	- What Nova errors look like
+		Nova errors will generally be the case where you are getting "No valid hosts available" when their should be.
+	- How to Fix
+		This will simply be the case where we see a dead service on some/all nodes.  Running through the checks will show any/all down nodes.  OpenStack won't deploy new VMs to nova-compute services that are down.  Attempting a restart should bring the services back online or show some errors in the logs about what is causing the nova-compute service to go down.
+	- How to take down nodes for maintenance
+		Eventually, you will have to take some compute nodes offline for maintenance in some capacity.  The first thing you want to do, is set those nova-computes to disabled:
+``` bash
+openstack compute service set --disable --disable-reason maintenance <node hostname> nova-compute
+```
+Then, check to see if the node has VMs running on it.  You can do this with:
+``` bash
+openstack host show <node hostname>
+```
+If this has any CPU/Memory/Disk used, it probably has VMs running on it and they need to be migrated to other nodes.  We can do this with:
+``` bash
+nova host-evacuate-live <node hostname>
+```
+This will send the VMs on that node to other availabe nodes.  Once this completes, the node will be out of production and available for the maintenance required.  After said maintenance is completed, you can bring the node back into production with:
```

Comment:
```suggestion
This will send the VMs on that node to other available nodes.  Once this completes, the node will be out of production and available for the maintenance required.  After said maintenance is completed, you can bring the node back into production with:
```

spelling fix

Also, we should tell people how to know when the evacuation completes. (Maybe by re-running `nova host show`?)